core/src/main/scala/ai/h2o/sparkling/backend/Writer.scala

```diff
@@ -134,7 +134,9 @@ private[backend] object Writer {
           con.putNA(idxField)
         } else {
           entry.dataType match {
-            case BooleanType => con.put(row.getBoolean(idxField))
+            case BooleanType =>
+              val valueIndex = domainBuilder.addStringToDomain(row.getBoolean(idxField).toString, idxField)
```

Comment:
It seems more natural to encode the boolean value as "0"/"1" instead of "true"/"false".

That would also be beneficial for backwards compatibility because if previously someone had a conversion code asfactor in their script they would end up with a domain represented as "0"/"1".